packages/@aws-cdk/aws-codepipeline-actions/lib/codecommit/source-action.ts

```diff
@@ -119,7 +122,7 @@ export class CodeCommitSourceAction extends Action {
     const createEvent = this.props.trigger === undefined ||
       this.props.trigger === CodeCommitTrigger.EVENTS;
     if (createEvent) {
-      this.props.repository.onCommit(stage.pipeline.node.uniqueId + 'EventRule', {
+      this.props.repository.onCommit(`${stage.pipeline.node.uniqueId}-${this.branch}-EventRule`, {
```

Comment:
Let's do this without breaking existing templates. It's a chore, but our customers will thank us.

Only insert the branch if its `!= 'master'` ?

(Not 100% guaranteed to not change anyone's templates, but like 90% good enough).

Reply:
Done!